addons/sourcemod/scripting/shavit-zones.sp

```diff
@@ -4319,6 +4437,33 @@ public void TouchPost(int entity, int other)
 	{
 		case Zone_Start:
 		{
+			if (gB_Eventqueuefix)
+			{
+				static int tick_served[MAXPLAYERS + 1];
+				int curr_tick = GetGameTickCount();
+				
+				// GAMMACASE: This prevents further abuses related to external events being ran after you teleport from the trigger, with events setup, outside the start zone into the start zone.
+				// This accounts for the io events that might be set inside the start zone trigger in OnStartTouch and wont reset them!
+				// Logic behind this code is that all events in this chain are not instantly fired, so checking if there were teleport from the outside of a start zone in last couple of ticks
+				// and doing PhysicsRemoveTouchedList() now to trigger all OnEndTouch that should happen at the same tick but later and removing them allows further events from OnStartTouch be separated
+				// and be fired after which is the expected and desired effect.
+				// This also kills all ongoing events that were active on the client prior to the teleportation to start and also resets targetname and classname
+				// before the OnStartTouch from triggers in start zone are run, thus preventing the maps to be abusable if they don't have any reset triggers in place
+				if((gI_LatestTeleportTick[other] <= curr_tick <= gI_LatestTeleportTick[other] + 4) && curr_tick != tick_served[other])
```

Comment:
what's the reason for the +4 tick range?

Reply:
Cuz the Touch function might skip few ticks when trigger is untouched, so in reality it needs more ticks to fully retrigger all triggers in start zone, and the 4 here was again fine tuned to be fine on the maps I have tested it. It all depends on the trigger call order and may vary from map to map.

Reply:
I'll try to get this merged in soon. This part is still just something I want to look at more though since +1 seems to work fine

Reply:
It does not on the bhop_overthinker (Only in CSS) and the test map I built, with multiple triggers inside each others. On bhop_overthinker that is what you were getting before I'm assuming, first time teleporting to the bonus there causes the events to not be refired (teleports within the bonus trigger works just fine and events would be applied). Also I don't really see how there additional 3 ticks might cause any issues, so if it adds the additional confidence that events would be refired, I'd leave that in.

Reply:
Ah, yep I see now. Going to do a commit for like two things and then squash and merge this